sympy/assumptions/cnf.py

```diff
@@ -0,0 +1,373 @@
+from itertools import combinations, product
+
+from sympy import S
+from sympy.logic.boolalg import BooleanFunction, Or, And, Not
+
+
+# Early encoding for literals
+literals_store = dict()
+
+
+class Literal(object):
+    """
+    The smallest element of a CNF object
+    """
+    literals_count = 0
+
+    def __new__(cls, lit, is_not=False):
+        if type(lit).__name__ == 'Not':
+            lit = lit.args[0]
+            is_not = True
+        k = literals_store.get(lit, None)
+        if k is not None:
+            code = k
+        elif lit in (S.false, False):
```

Comment:
`False` already compares equal to `S.false`, so you can just use `lit == S.false` here. Unless `0` is supposed to match as well (but unless I am mistaken I think it shouldn't).